packages/js/product-editor/src/utils/parse-product-to-blocks.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import { Product } from '@woocommerce/data';
+import { BlockInstance, createBlock } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import nameBlock from '../components/details-name-block/block.json';
+
+export function parseProductToBlocks( product: Partial< Product > ) {
+	const blocks: BlockInstance[] = [];
+
+	blocks.push(
+		createBlock( nameBlock.name, {
+			name: product.name,
```

Comment:
Just curious if this logic will still be needed if we start using templates to generate the form?

Reply:
After our conversation, my thought is no. I believe once we have [your template PR merged](https://github.com/woocommerce/woocommerce/pull/37123) I can just register this block and move it to the template..? If you get that merged first, I'm glad to rebase and retool a bit. Otherwise we can migrate it in that PR or another follow-up.

Reply:
Also in favor of removing this.  Seems like at best we can avoid this, but worst case is we will need to parse the product and retrofit it into the hydrated template.

Reply:
Louren's settings PR is now merged, and I've rebased this PR to include it. I started experimenting with loading the blocks from the post type template in d284e9b82c4ab915a3eb680a8d1efee2bad85486. I'm uncertain of the pattern here, and arguably this might belong in a separate PR. I can continue on Monday but let me know if you have any thoughts @louwie17

Reply:
Hmm I don't think its necessary to register the template as a block (unless you saw a similar pattern in Gutenberg somewhere?)
In theory we should just be able to pass it into the `setupEditor` function similar to what Josh started doing here: https://github.com/woocommerce/woocommerce/pull/36921/files#diff-5fbb147f0c06b95b81afa9f70ae6db13e289b2714f4b21d041a7bd88670a1d03R106

Josh's approach is similar to Gutenberg's approach here: https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/provider/index.js#L84
We may also want to make use of the `useBlockEditorSettings` that is being made use of there (if the call to `setupEditor` is not enough).

Reply:
But on second thought it might help to leave this as is for now and merge it, to un-block any block field work and create a separate PR for the `template` support.